kratos/solving_strategies/convergencecriterias/mixed_generic_criteria.h

```diff
@@ -0,0 +1,441 @@
+        GetNormValues(rModelPart, rDofSet, rDx, dofs_count, solution_norms_vector, increase_norms_vector);
+
+        // Synchronize the norm values
+        const auto& r_data_comm = rModelPart.GetCommunicator().GetDataCommunicator();
+        auto global_solution_norms_vector = r_data_comm.SumAll(solution_norms_vector);
+        auto global_increase_norms_vector = r_data_comm.SumAll(increase_norms_vector);
+        auto global_dofs_count = r_data_comm.SumAll(dofs_count);
+
+        // Check division by zero in global solution norms
+        const double zero_tol = 1.0e-12;
+        for(int i = 0; i < mVariableSize; i++) {
+            if (global_solution_norms_vector[i] < zero_tol) {
+                global_solution_norms_vector[i] = 1.0;
+            }
+        }
+
+        // Calculate the norm values
+        std::vector<TDataType> var_ratio(mVariableSize, 0.0);
+        std::vector<TDataType> var_abs(mVariableSize, 0.0);
+        for(int i = 0; i < mVariableSize; i++) {
+            var_ratio[i] = std::sqrt(global_increase_norms_vector[i] / global_solution_norms_vector[i]);
+            var_abs[i] = std::sqrt(global_increase_norms_vector[i]) / static_cast<TDataType>(global_dofs_count[i]);
+        }
+
+        // Output the ratio and absolute norms as a tuple
+        return std::make_tuple(var_ratio, var_abs);
+    }
+
+    /**
+     * @brief Method to output the convergence status
+     * This method prints the convergence status to the screen for each one of the checked variables
+     * @param rConvergenceNorms Tuple containing the absolute and relative convergence values
+     */
+    void OutputConvergenceStatus(const std::tuple<std::vector<TDataType>, std::vector<TDataType>>& rConvergenceNorms)
+    {
+        const auto& var_ratio = std::get<0>(rConvergenceNorms);
+        const auto& var_abs = std::get<1>(rConvergenceNorms);
+
+        if (this->GetEchoLevel() > 0) {
+            std::ostringstream stringbuf;
+            stringbuf << "CONVERGENCE CHECK:\n";
+            for(int i = 0; i < mVariableSize; i++) {
+                const auto r_var_data = mVariableDataVector[i];
+                const int key_map = mLocalKeyMap[r_var_data->Key()];
+                stringbuf << " " << r_var_data->Name() << " : ratio = " << var_ratio[key_map] << "; exp.ratio = " << mRatioToleranceVector[key_map] << " abs = " << var_abs[key_map] << " exp.abs = " << mAbsToleranceVector[key_map] << "\n";
+            }
+            KRATOS_INFO("") << stringbuf.str();
+        }
+    }
+
+    /**
+     * @brief Method to check convergence
+     * This method checks the convergence of the provided norms with the user-defined tolerances
+     * @param rConvergenceNorms Tuple containing the absolute and relative convergence values
+     * @return true Convergence is satisfied
+     * @return false Convergence is not satisfied
+     */
+    bool CheckConvergence(const std::tuple<std::vector<TDataType>, std::vector<TDataType>>& rConvergenceNorms)
```

Comment:
```suggestion
    bool CheckConvergence(const std::tuple<std::vector<TDataType>, const std::vector<TDataType>>& rConvergenceNorms) const
```

Reply:
Impossible because of the `operator[]` of the map.